### ISSUE TITLE
azuread_service_principal_password - end_date

### ISSUE THREAD

**Issue by aleqsss** [hashicorp/terraform-provider-azuread#701]
Since updating to version 2.0.0 of the provider (I guess that you updated to use the Microsoft Graph API instead of the deprecated Azure Active Directory Graph API), it's not possible to specify any **end_date** for the **azuread_service_principal_password** resource. It automatically defaults to two years.

I can see that it's still possible for the **azuread_application_password** resource though, how come it's possible for that resource, but not for the **azuread_service_principal_password**?

When creating a service principal via the Azure portal/Azure CLI/PowerShell (New-AzADServicePrincipal), you have the option to specify the end date of the password.

Why is this not present in this provider and will it be? Is there any other way to change the end date of the **azuread_service_principal_password**, from within the Terraform configuration?

Thank you!

**Comment by manicminer** [comment]
Hi @aleqsss, thanks for requesting this. At the time 2.0.0 was released, and for a period afterwards, the API exhibited a bug where the display name, start date and expiry date for service principal passwords was simply ignored and set to default values on creation. Unfortunately this meant we had to remove these properties from the resource as they were having no effect.

Although I revisited this a couple times without success, I have retested this now and it appears the API is now working as documented! As such, it looks like we'll be able to re-add these in the provider.
